TopMovies/Resources/Assembler/Assembler.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  Assembler.swift
+//  TopMovies
+//
+//  Created by Macbook Pro  on 19.03.2021.
+//
+
+import Swinject
+import ReSwift
+import Overture
+
+typealias MainStore = Store<MainState>
+
+struct Assembler {
+  static func registerDependencies() {
+    let registrar = Container.default.registrar
+    
+    // MARK: - Store
+    registrar.register(MainStore.self) { _ in
+      MainStore(reducer: mainReducer, state: nil, middleware: [allActionsMiddleware])
+    }.inObjectScope(.container)
+    
+    // MARK: - Service
+    registrar.register(MovieAPIProtocol.self) { _ in MovieAPI() }
+    registrar.register(MovieService<StoreProvider<MainState>>.self) { resolver in
+      MovieService(
+        movieAPI: resolver.resolve(MovieAPIProtocol.self)!,
+        storeProvider: curry(StoreProvider<MainState>.init(store: onStateUpdate:))(resolver.resolve(MainStore.self)!)
+      )
+    }.inObjectScope(.container)
+    
+    // MARK: - Factory
+    registrar.register(VCFactoryProtocol.self) { _ in
+      VCFactory()
+    }
+    registrar.register(ConnectorFactoryProtocol.self) { _ in
+      ConnectorFactory()
+    }
+    
+    // MARK: - Router
+    registrar.register(RouterProtocol.self) { resolver in
+      Router(vcFactory: resolver.resolve(VCFactoryProtocol.self)!,
+             connectorFactory: resolver.resolve(ConnectorFactoryProtocol.self)!)
```

Comment:
Can we just inject them inside of the router?

Reply:
Yes, but i'm really don't sure about DI by @Inject property wrapper, it's good enough for some global things like Store or Service for usage in code, but not for resolving dependencies inside objects as for me

Reply:
You still contain those factories in Container, so it doesn't make sense to resolve them with force-unwrap inside of init. That syntax doesn't looks clean as for me and it doesn't make any difference, since you can always resolve any other factory in container to perform your taks